src/App.test.js

```diff
@@ -1,8 +1,15 @@
-import { render, screen } from '@testing-library/react';
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { BrowserRouter as Router } from 'react-router-dom';
 import App from './App';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+test('Whole Code Snapshot testing', () => {
+  const components = renderer.create(
+    <Router>
+      <App />
+    </Router>,
+
+  );
+  const tree = components.toJSON();
+  expect(tree).toMatchSnapshot();
 });
```

Comment:
- [ ] Kindly add unit tests for all components. Currently, you are missing tests for the following components: `Calculator`, `Header`, `Home`, and `Quotes`. Kindly add them.

- [ ] You are also missing tests that simulate user interactions like clicking buttons and expectations. Kindly add those as well e.g. performing a calculation on the calculator component.